### ISSUE TITLE
False Negative in LeftCurly for ENUM_DEF

### ISSUE THREAD

**Issue by Atharv3221** [checkstyle/checkstyle#20956]
I have read check documentation: https://checkstyle.sourceforge.io/checks/blocks/leftcurly.html#LeftCurly
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```java
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat Test.java
enum Test { ONE,
    TWO,
}
```
With config1:
```
<module name="Checker">
  <module name="TreeWalker">
    <module name="LeftCurly">
      <property name="option" value="eol"/>
      <property name="ignoreEnums" value="false"/>
      <property name="tokens" value="ENUM_DEF, OBJBLOCK"/>
    </module>
  </module>
</module>
```
```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-13.8.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/atharv/test/Test.java:1:11: '{' at column 11 should have line break after. [LeftCurly]
Audit done.
Checkstyle ends with 1 errors.
```

With config2:
```
<module name="Checker">
  <module name="TreeWalker">
    <module name="LeftCurly">
      <property name="option" value="eol"/>
      <property name="ignoreEnums" value="false"/>
      <property name="tokens" value="ENUM_DEF"/>
    </module>
  </module>
</module>
```

```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-13.8.0-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

Expected:
```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-13.8.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/atharv/test/Test.java:1:11: '{' at column 11 should have line break after. [LeftCurly]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by Atharv3221** [comment]
I am working on this

**Comment by romani** [comment]
may be OBJBLOCK is technical token that user should not be able to configure, as it is present for all types and synonym for LCURLY, right ?

**Comment by Atharv3221** [comment]
@romani 
no we require OBJBLOCK token for anonymous class
```java
public class Test {
  public void test() {
    Greeter g = new Greeter() {

    };
  }
}
```
```
        |       |--VARIABLE_DEF -> VARIABLE_DEF [3:5]
        |       |   |--MODIFIERS -> MODIFIERS [3:5]
        |       |   |--TYPE -> TYPE [3:5]
        |       |   |   `--IDENT -> Greeter [3:5]
        |       |   |--IDENT -> g [3:13]
        |       |   `--ASSIGN -> = [3:15]
        |       |       `--EXPR -> EXPR [3:17]
        |       |           `--LITERAL_NEW -> new [3:17]
        |       |               |--IDENT -> Greeter [3:21]
        |       |               |--LPAREN -> ( [3:28]
        |       |               |--ELIST -> ELIST [3:29]
        |       |               |--RPAREN -> ) [3:29]
        |       |               `--OBJBLOCK -> OBJBLOCK [3:31]
        |       |                   |--LCURLY -> { [3:31]
        |       |                   `--RCURLY -> } [5:5]
        |       |--SEMI -> ; [5:6]

```

**Comment by romani** [comment]
ok OBJBLOCK is for anonymous.

https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ENUM_DEF 
also has:
```
  `--OBJBLOCK -> OBJBLOCK
      |--LCURLY -> {
      |--ENUM_CONSTANT_DEF -> ENUM_CONSTANT_DEF
```

so looks like it is required token for ENUM_DFF to work.

be care full that `tokens` property is very technical, most user do not understand it as it requires knowledge of AST, that nobody want to deal with. So it is very technical requirements, lets keep it as it. So reason to change.

**Comment by Atharv3221** [comment]
Lets see third case:

```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat Test.java
enum Test { ONE,
    TWO,
}
```
With config
```
<module name="Checker">
  <module name="TreeWalker">
    <module name="LeftCurly">
      <property name="option" value="eol"/>
      <property name="ignoreEnums" value="false"/>
      <property name="tokens" value="OBJBLOCK"/>
    </module>
  </module>
</module>
```
```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-13.8.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/atharv/test/Test.java:1:11: '{' at column 11 should have line break after. [LeftCurly]
Audit done.
Checkstyle ends with 1 errors.
```

Violation reported without using ENUM_DEF token, I don't see why user will expect violation if they explicitly didn't configured ENUM_DEF

Problem is check is validating enum based on OBJBLOCK not ENUM_DEF which is incorrect

**Comment by Atharv3221** [comment]
similar issue https://github.com/checkstyle/checkstyle/issues/6459
